ejb3/src/main/java/org/jboss/as/ejb3/remote/AssociationImpl.java

```diff
@@ -217,6 +230,34 @@ public CancelHandle receiveInvocationRequest(@NotNull final InvocationRequest in
         return cancellationFlag::cancel;
     }
 
+    private static <A> boolean intersect(final Set<A> fst, final Set<A> snd) {
+        fst.retainAll(snd);
+        return !fst.isEmpty();
+    }
+
+    private static Set<String> typesOf(final Object o) {
```

Comment:
@dmlloyd makes sense for sure. In this case we will fix Weld part first and then merge this PR, in which there would be no exclusions at all, I suppose.
Remaining question is what else is being put in the map by ejb30/bb/session/stateful/interceptor/listener/override that causes TCK to fail. If you have no suggestions/time at the moment I will try debugging myself and create issues and link them accordingly.